packages/sdk/client-services/src/packlets/vault/iframe-compatibility-runtime.ts

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Trigger } from '@dxos/async';
+import { Config } from '@dxos/config';
+import { logInfo } from '@dxos/log';
+import { MemorySignalManager, MemorySignalManagerContext, WebsocketSignalManager } from '@dxos/messaging';
+import { createWebRTCTransportFactory, NetworkManager } from '@dxos/network-manager';
+import { createProtoRpcPeer, ProtoRpcPeer, RpcPort } from '@dxos/rpc';
+import { MaybePromise } from '@dxos/util';
+
+import { clientServiceBundle, ClientServices, ClientServicesHost } from '../services';
+
+export type IFrameCompatibilityRuntimeParams = {
+  configProvider: () => MaybePromise<Config>;
+  appPort: RpcPort;
+};
+
+export class IFrameCompatibilityRuntime {
```

Comment:
IFrameHostRuntime

Reply:
Comment please. What, how, when? Really important and complex class.